httpcore5/src/main/java/org/apache/hc/core5/http/impl/io/HttpRequestExecutor.java

```diff
@@ -212,7 +213,7 @@ public ClassicHttpResponse execute(
             return response;
 
         } catch (final HttpException | IOException | RuntimeException ex) {
-            Closer.closeQuietly(conn);
```

Comment:
@ooeunz Let's do better. Could you please do `CloseMode.IMMEDIATE` for `IOException`s and `RuntimeException`s and regular close for `SSLException`s and `HttpException`s?

Reply:
@ok2c Sure, but since SSLException is a subclass of IOException, I think it should be worded like this.
```
        } catch (final IOException | RuntimeException ex) {
            Closer.close(conn, CloseMode.IMMEDIATE);
            throw ex;
        } catch (final HttpException ex) {
            Closer.closeQuietly(conn);
            throw ex;
        }
```

Reply:
@ooeunz How about this?
```
        } catch (final HttpException | SSLException ex) {
            Closer.closeQuietly(conn);
            throw ex;
        } catch (final IOException | RuntimeException ex) {
            Closer.close(conn, CloseMode.IMMEDIATE);
            throw ex;
        }
```

Reply:
@ok2c Thank you, I misunderstood your intention. I’ve updated the code accordingly. :)